docs/CANBus.md

```diff
@@ -0,0 +1,391 @@
+
+## Device model
+
+CAN interfaces (physical or virtual) are specified in the Device Model
+file as members of the list of I/O devices (_ioMemberList_), for instance, a
+physical CAN interface (_can0_) can be configured by the following code:
+
+```json
+"ioMemberList": [
+    {
+        "ztype": "IO_TYPE_CAN",
+        "phylabel": "can0",
+        "logicallabel" : "can0",
+        "assigngrp" : "",
+        "phyaddrs" : {
+            "ifname" : "can0"
+        },
+        "cbattr" : {
+            "bitrate": "125000",
+            "sample-point": "0.875",
+            "tq": "29",
+            "prop_seg": "118",
+            "phase_seg1": "119",
+            "phase_seg2": "34",
+            "sjw": "1",
+            "dbitrate": "250000",
+            "dsample-point": "0.875",
+            "dtq": "29",
+            "dprop_seg": "118",
+            "dphase_seg1": "119",
+            "dphase_seg2": "34",
+            "dsjw": "1",
+            "loopback": "on",
+            "listen-only": "off",
+            "triple-sampling": "off",
+            "one-shot": "off",
+            "berr-reporting": "on",
+            "fd": "on",
+            "fd-non-iso": "off",
+            "presume-ack": "on",
+            "cc-len8-dlc": "off",
+            "tdc-mode": "auto"
+        }
+    }
+]
+```
+
+The parameter _ztype_ shall be:
+
+* __IO_TYPE_CAN__: For physical CAN interfaces
+
+* __IO_TYPE_VCAN__: For virtual CAN interfaces
+
+* __IO_TYPE_LCAN__: For a logical CAN device, i.e., this type of device doesn’t
+  define any configuration for a CAN controller, but it just points to
+  another (physical or virtual) CAN interface. It should be particularly
+  used to duplicate a CAN interface in the device model so it can be
+  passed-through to more than one Edge Application
+
+Only the following parameters are mandatory:
+
+* ztype
+
+* phylabel
+
+* logicallabel
+
+* cbattr (at least bitrate should be present for IO_TYPE_CAN type)
+
+Parameters not supported by the CAN interface shall be omitted from the device model file.
+
+### Giving access to host's CAN controllers to Edge Applications
+
+It’s important to point out that in case of CAN interfaces, the same CAN
+interface can be assigned to one or more Edge Applications. This
+requirement breaks the logic for device passthrough on EVE + Cloud
+Controller. So the alternative approach is to use the logical CAN
+definition, which points to a physical/virtual CAN interface in the device
+model. For instance, the use case from Fig. 2 would have the following
+configuration in the device model:
+
+```json
+"ioMemberList": [
+    {
+        "ztype": "IO_TYPE_CAN",
+        "phylabel": "can0",
+        "logicallabel": "can0",
+        "assigngrp" : "",
+        "phyaddrs" : {
+            "ifname" : "can0"
```

Comment:
Is reference by interface name reliable for CAN devices? Or could the init order of CAN devices and thus their interface names change between reboots?

Reply:
yes, it is. In case of CAN there is no other possible association, like MAC addresses, serial numbers... perhaps we could look some other attribute for physical controllers, but definitely not for virtual CAN, so the ifname is the common one for all interfaces. Also, now that we have udev a eventual problem could be solved with udev rules, right?